GPyOpt/testing/methods_tests/test_bayesian_optimization.py

```diff
@@ -35,4 +35,16 @@ def test_next_locations_ignored(self):
         bo_ignored = BayesianOptimization(f = None, domain = domain, X = X_init, Y = Y_init, de_duplication = True)
         x_ignored = bo_ignored.suggest_next_locations(ignored_X = x_no_ignored)
 
-        self.assertFalse(np.isclose(x_ignored, x_no_ignored))
\ No newline at end of file
+        self.assertFalse(np.isclose(x_ignored, x_no_ignored))
+
+    def test_one_initial_data_point(self):
+        """Make sure BO still works with only one initial data point."""
+        bounds = [{'name': 'var_1', 'type': 'continuous', 'domain': (-1, 1)}]
+        opt = BayesianOptimization(lambda x: x, bounds, initial_design_numdata=1)
+
+        # Make sure the distance function doesn't fail quietly
+        self.assertRaises(IndexError, opt._distance_last_evaluations)
```

Comment:
Thanks for the test. One last thing: I believe this line should not be here, as `_distance_last_evaluations` is considered internal for the BO, and ideally should not be used by the unit tests. Can you please remove it?

If there is a way to trigger this IndexError using API of the BO class, we could have another test for that, but at the moment i can't think of one

Reply:
Would be a pretty bad PR if this was possible :)

I've removed the internal test